base/strings/string.jl

```diff
@@ -60,7 +60,35 @@ This representation is often appropriate for passing strings to C.
 String(s::AbstractString) = print_to_string(s)
 String(s::Symbol) = unsafe_string(unsafe_convert(Ptr{UInt8}, s))
 
-(::Type{Vector{UInt8}})(s::String) = ccall(:jl_string_to_array, Ref{Vector{UInt8}}, (Any,), s)
+unsafe_wrap(::Type{Vector{UInt8}}, s::String) = ccall(:jl_string_to_array, Ref{Vector{UInt8}}, (Any,), s)
+
+"""
+    StringBytes(s::String)
+
+Wrap a `String` (without copying) in an immutable vector-like object that accesses the bytes
+of the string's representation.
+"""
+struct StringBytes <: AbstractVector{UInt8}
```

Comment:
`<: DenseVector{UInt8}`, and include `strides` and `unsafe_convert(Ptr{UInt8}, ...)` methods?